pkg/trace/writer/sender.go

```diff
@@ -272,6 +276,15 @@ func (s *sender) sendPayload(p *payload) {
 	}
 }
 
+// shouldWarnRetry determines whether a warning should be emitted
+// with the given count of retries to avoid spam
```

Comment:
```suggestion
// shouldWarnRetry determines whether a warning should be emitted
// after it has been retried n times.
```
It's nice to try and use the actual argument name in the documentation. It is a bit more friendly and I noticed that's how most repositories do it, including the standard library.

Also, I think in this case `n` makes more sense than `r` as a count.